nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/dimensionalityreduction/PCA.java

```diff
@@ -169,7 +169,7 @@ public static INDArray pca(INDArray A, int nDims, boolean normalize) {
      * @return the reduced feature set
      */
     public static INDArray pca_factor(INDArray A, int nDims, boolean normalize) {
-
+        A = A.dup();
```

Comment:
I haven't had the time to review this earlier, but I am afraid that this is not going to solve the issue where A is modified after the call further down in:
`Nd4j.getBlasWrapper().lapack().gesvd(A, s, null, VT);`
`A = A.dup();` just duplicates A and assigns it back to A, so after the call to gesvd A will be modified.
The only way to leave A unmodified after the call to gesvd is to do the following:
`Nd4j.getBlasWrapper().lapack().gesvd(A.dup(), s, null, VT);`
That way gesvd operates and modifies the duplicate and leaves A intact for further use.

Reply:
> A = A.dup(); just duplicates A and assigns it back to A, so after the call to gesvd A will be modified.

Yes, but no. As the reference to A gets changed, it means that A inside the method gets changed, but the A that was originally passed to pca_factor will not be changed.

Reply:
You are right, my paranoia from C-like languages got better of me. I never assign to input parameters even when it is OK to do it like here.